doc/api/esm.md

```diff
@@ -647,21 +651,24 @@ export async function resolve(specifier, context, defaultResolve) {
 }
 ```
 
-#### `getFormat(url, context, defaultGetFormat)`
+### `load(url, context, defaultLoad)`
 
-> Note: The loaders API is being redesigned. This hook may disappear or its
-> signature may change. Do not rely on the API described below.
+> Note: Previously the experimental version had 3 separate hooks, which have
+> been consolidated into this one.
 
 * `url` {string}
 * `context` {Object}
-* `defaultGetFormat` {Function}
+* `defaultLoad` {Function}
 * Returns: {Object}
   * `format` {string}
+  * `source` {string|SharedArrayBuffer|Uint8Array}
+
+The `load` hook provides a way to define a custom method of determining how
+a URL should be interpreted, retrieved, and parsed.
 
-The `getFormat` hook provides a way to define a custom method of determining how
-a URL should be interpreted. The `format` returned also affects what the
-acceptable forms of source values are for a module when parsing. This can be one
-of the following:
+For chained loaders, intermediary values of `format` can be any string (perhaps
+to be used by another loader in the chain); however, the _final_ value of
+`format` must be one of the following:
```

Comment:
Given that this PR implements chaining, we should make sure to properly cover chaining semantics here as well in terms of how multiple loaders treat `defaultLoad` and what chaining patterns apply.

Reply:
This PR doesn't fully cover chaining (it merely lays the groundwork for it). There're two artificial gates that actually prevent it:

* `--experimental-loader` does not accept an array (partially because I dunno where/how to update it to accept multiple arguments, partially because it's not supposed to yet); specifying more than 1 via CLI causes subsequent values to overwrite previous (last-wins).
* `ESMLoader.resolve()` & `ESMLoader.load()` use specifically only the first (default) hook and the second (custom) hook; _if_ a user was somehow able to shove more custom hooks in, they would just get ignored.

How chaining is actually going to work is still TBD: #36954. This PR leans toward option `next()` (partially because it did before), buuuut in writing the code and tests for it, I was further inclined away from the `next()` approach (see [my comment](https://github.com/nodejs/node/issues/36954#issuecomment-822097316) on 36954).